client/src/modules/authorization/components/PasswordRecoveryForm.jsx

```diff
@@ -0,0 +1,227 @@
+import { updatePassword } from '../model/actions';
+import ValidationInput from './ValidationInput';
+import PendingButton from '../../../common/components/PendingButton';
+
+class PasswordRecoveryForm extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: {
+        comparePasswordsError: '',
+        passwordError: '',
+        passwordUpdateError: false
+      },
+      password: '',
+      passwordConfirmation: '',
+      passwordConfirmationSuccess: false,
+      passwordSuccess: false,
+      pending: false
+    };
+
+    this.debouncedChange = undefined;
+  }
+
+  handlePasswordChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.debouncedChange = _debounce(() => this.validatePassword(value), 500);
+    this.setState({ password: value }, this.debouncedChange);
+  };
+
+  handlePasswordConfirmationChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.debouncedChange = _debounce(() => this.comparePasswords(), 500);
+    this.setState({ passwordConfirmation: value }, this.debouncedChange);
+  };
+
+  handleSubmit = () => {
+    const { password } = this.state;
+    const {
+      match: {
+        params: { token }
+      }
+    } = this.props;
+
+    this.setState({ pending: true });
+    updatePassword(token, password)
+      .then(() => this.setState({ pending: false }))
+      .catch(() => {
+        const { errors } = this.state;
+
+        this.setState({
+          errors: {
+            ...errors,
+            passwordUpdateError: true
+          },
+          password: '',
+          passwordConfirmation: '',
+          passwordConfirmationSuccess: false,
+          passwordSuccess: false,
+          pending: false
+        });
+      });
+  };
+
+  validatePassword = password => {
+    const { errors, passwordConfirmation } = this.state;
+    const { matches } = validator;
+    const validationError = !matches(password, /^[^\s]{4,32}$/);
+    const passwordMinLength = password.length >= 4;
+
+    if (passwordConfirmation) {
+      this.comparePasswords();
+    }
+
+    if (_isEmpty(password)) {
+      return this.setState({
+        errors: {
+          ...errors,
+          comparePasswordsError: '',
+          passwordError: ''
+        },
+        passwordSuccess: false
+      });
+    }
+
+    if (validationError) {
+      return this.setState({
+        errors: {
+          ...errors,
+          passwordError: 'authorization.input.password.invalid'
+        },
+        passwordSuccess: false
+      });
+    }
+
+    if (passwordMinLength && !validationError) {
+      this.setState({
+        errors: {
+          ...errors,
+          passwordError: ''
+        },
+        passwordSuccess: true
+      });
+    }
+  };
+
+  comparePasswords = () => {
+    const { errors, password, passwordConfirmation } = this.state;
+    const passwordMinLength = password.length >= 4;
+
+    if (
```

Comment:
maybe just following check:
```
if (password.length >= 4 && _trim(password) === _trim(passwordConfirmation)) {
    // pasword OK
} else {
    // password notOK
}
```

Reply:
Okay I can merge those two if statements within if else block, 
but anyway I still need third IF, beacuse I have 3 diffrent states here: 
- no error, passwordConfirmationSuccess: true
- no error, passwordConfirmationSuccess: false,
- error, passwordConfirmationSuccess: false

I can't handle them in just two logical blocks cause I need to check if confirmationPassword is empty, otherwise I will get this situation:
<img width="599" alt="Zrzut ekranu 2019-07-04 o 07 35 24" src="https://user-images.githubusercontent.com/27632432/60641942-5566de80-9e2e-11e9-8ab2-8f95b6dcd9b0.png">